javascript/signup.js

```diff
@@ -1,67 +1,88 @@
+    else if(!isValidEmail(inputEmail.value)) {
+        showErrorMessage(inputBoxEmail, "올바른 이메일 주소가 아닙니다.");
     }
-    else if(sign.inputEmail.value === sign.USER_EMAIL) {
-        sign.showErrorMessage(sign.inputBoxEmail, "이미 사용 중인 이메일입니다.");
+    else if(inputEmail.value === USER_EMAIL) {
+        showErrorMessage(inputBoxEmail, "이미 사용 중인 이메일입니다.");
     }
     // error message remove
-    else if(sign.isValidEmail(sign.inputEmail.value)) {
-        sign.removeMessage(sign.inputBoxEmail);
+    else if(isValidEmail(inputEmail.value)) {
+        removeMessage(inputBoxEmail);
+        checkUsableEmail(inputEmail.value);
     }
 }
 // focus-out password
 function eventFocusOutPassword() {
     // no password value
-    if(sign.inputPassword.value === "") {
-        sign.showErrorMessage(sign.inputBoxPassword, "비밀번호를 입력해주세요");
+    if(inputPassword.value === "") {
+        showErrorMessage(inputBoxPassword, "비밀번호를 입력해주세요");
     }
-    else if(!isValidPassword(sign.inputPassword.value)) {
-        sign.showErrorMessage(sign.inputBoxPassword, "비밀번호는 영문, 숫자 조합 8자 이상 입력해주세요.");
+    else if(!isValidPassword(inputPassword.value)) {
+        showErrorMessage(inputBoxPassword, "비밀번호는 영문, 숫자 조합 8자 이상 입력해주세요.");
     }
     else {
-        sign.removeMessage(sign.inputBoxPassword);
+        removeMessage(inputBoxPassword);
     }
 }
 function eventFocusOutPasswordCheck() {
     // no password value
-    if(inputPasswordCheck.value !== sign.inputPassword.value) {
-        sign.showErrorMessage(inputBoxPasswordCheck, "비밀번호가 일치하지 않아요.");
+    if(inputPasswordCheck.value !== inputPassword.value) {
+        showErrorMessage(inputBoxPasswordCheck, "비밀번호가 일치하지 않아요.");
     }
     else {
-        sign.removeMessage(inputBoxPasswordCheck);
+        removeMessage(inputBoxPasswordCheck);
     }
 }
 // signup button click
-function eventClickBtnSignup() {
-    if( sign.isValidEmail(sign.inputEmail.value)
-        && isValidPassword(sign.inputPassword.value)
-        && inputPasswordCheck.value === sign.inputPassword.value ) {
+async function eventClickBtnSignup() {
+    try {
+        // 이메일 또는 패스워드 형식에 맞지 않는 값 입력시 에러 처리
+        if( !isValidEmail(inputEmail.value)
+        || !isValidPassword(inputPassword.value)
+        || !(inputPasswordCheck.value === inputPassword.value) ) {
+            throw new Error("unusable sign-up form.");
+        }
+
+        // POST 메소드로 가입 가능 여부 체크
+        const signupForm = {
+            email: inputEmail.value,
+            password: inputPassword.value
+        };
+        const response = await fetch('https://bootcamp-api.codeit.kr/api/sign-up', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(signupForm),
+        })
+        if (response.ok) {
+            const result = await response.json();
+            localStorage.setItem('accessToken', result['data']['accessToken']);
```

Comment:
c: 로컬스토리지의 accessToken이라는 키에 레코드를 저장하는 로직이 반복되고 있네요. 공통화할 수 있을까요?

```suggestion
            localStorage.setItem('accessToken', result['data']['accessToken']);
```